deepgram/__init__.py

```diff
@@ -9,18 +9,25 @@
 from .scopes import Scopes
 from .invitations import Invitations
 from .extra import Extra
+from .errors import DeepgramSetupError, DeepgramApiError
 
 
 class Deepgram:
     def __init__(self, options: Union[str, Options]) -> None:
-        if not isinstance(options, str) and not options.get('api_url'):
-            raise ValueError("DG: API URL must be valid or omitted")
-        t_options: Options = {
-            'api_key': options
-        } if isinstance(options, str) else options
-        if 'api_key' not in t_options:
-            raise ValueError("DG: API key is required")
-        self.options = t_options
+        if not isinstance(options, (str, dict)):
+            raise DeepgramSetupError("`options` must be a dictionary or an API key string")
+
+        # Convert to dictionary if the api key was passed as a string
+        if isinstance(options, str):
+            options: Options = {"api_key": options}
+
+        if 'api_key' not in options:
+            raise DeepgramSetupError("API key is required")
+
+        if "api_url" in options and options.get("api_url", None) is None:
```

Comment:
I completely agree. The existing code has a mix of single quotes and double quotes. Python's standard is double quotes, so that's what I use. I did not want to add a ton of formatting changes to this PR, so I kept them minimal. I can change this one in particular because you raised it, but I'd like to go through and reformat the entire codebase in a later PR.